cloudimg/aws.py

```diff
@@ -357,7 +396,10 @@ def publish(self, metadata):
             An EC2 Image
         """
         log.info('Searching for image: %s', metadata.image_name)
-        image = self.get_image_by_name(metadata.image_name)
+        image = (
+            self.get_image_by_name(metadata.image_name) or
+            self.get_image_by_tags(metadata.tags)
```

Comment:
I'm concerned about this bit.

A summary of what it means is: if two images have the same (non-empty) tags, they should be considered the same image.  That seems correct if and only if every image with non-empty tags is expected to have unique tags.

Can you explain a bit (or link to a doc explaining) the usage of tags and where they come from?

Reply:
Yes, the idea is that we'll have 6 tags to uniquely identify an image in our workflow.

After uploading the image we'll tag it with the following values (6 tags in total):

- nvra: “Full NVRA” in environments that support it
- name: “Name” from the build’s NVR.
-version: “Version” from the build’s NVR.
- release: “Release” from the build’s NVR.
- arch: Image architecture.
Note: We need this because one build (one NVR) in Koji may contain multiple images each for a different architecture.
- buildid: ID of build in Koji.